### PR TITLE
Add APNs environment information

### PR DESCRIPTION
**Pull request by jcaron23**
## Description

```
Read the embedded.mobileprovision file and retrieve the
'aps-environement' entitlement to determine whether we're using the
development (sandbox) environment or the production environment.

Add "apsEnvironment" (string), "apsIsProduction" (boolean), and
"apsIsDevelopment"  (boolean)keys to the object passed to the
"registration" event handler.

One can then pass the information to the backend server to use the
correct environment (certificate, key, host).
```

Updated documentation.
## Motivation and Context

To send pushes through APNs, one needs the token but also the environment. The current implementation only returns the former, this change adds the latter.
## How Has This Been Tested?

Tested in iOS app on iOS 9.3.2.
## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: docs/API.md [hunk 1]
```diff
@@ -148,6 +148,9 @@ The event `registration` will be triggered on each successful registration with
 Parameter | Type | Description
 --------- | ---- | -----------
 `data.registrationId` | `string` | The registration ID provided by the 3rd party remote push service.
+`data.apsEnvironment` | `string` | (iOS only): the `aps-environement` entitlement in the `embedded.mobileProvision` file included in the app's bundle. Values can be `development` (sandbox) or `production`. You can pass this to your back-end server together with the registrationID to identify the environment (certificate, key, host) to use.
+`data.apsIsProduction` | `boolean` | (iOS only): `true` if `data.apsEnvironment` is `production`.
```
Comment (fredgalvao) [inline docs/API.md]: `data.apsIsProduction` and `data.apsIsDevelopment`, being direct products of `data.apsEnvironment`, seem unnecessary to me.

**Comment by macdonst** [comment]
@jcaron23 thanks for the PR I will end up merging this next week and it looks like it can be used to remove the need for ios.gcmSandbox.

**Comment by briananderson1222** [comment]
I think this PR might be somewhat of a requirement for building apps with XCode 8

**Comment by macdonst** [comment]
@briananderson1222 well, the PR has never compiled cleanly for me. I just tried again and it still doesn't in Xcode 8.

@jcaron23 any chance of you taking another look at this PR?

**Comment by jimduchek** [comment]
I just did a push request that's based off this patch (I'm in a hurry and I am not sure how to mess with someone elses push request -- it was easier to just do it separate).

**Pull request closed without merging**